### DISCUSSION TITLE
help me

### DISCUSSION THREAD

**Discussion by nadongjin** [leondgarse/Keras_insightface#60] [Q&A] [answered]
Hello
I have a question.
Personally, is there a way to detect faces in images I have, like glint360k or ms1m ?
mtcnn, retina, scrfd, etc. hold too tight when detecting faces.
thank you.

**Comment by leondgarse** [comment, marked as answer]
Ya, mostly detectors will not work well on those extracted training face images. I'm considering if I can hold a yolov5-face model here, but that may also not work well. May I know why you need that?

**Reply by nadongjin** [reply]
Hello
I am learning deepglint360k dataset.
Also, I have an additional data set.
I want to train with deepglint360k + additional dataset , but I don't know how to get a roi like deepglint360k.
One way I think of it is to get a face roi through face detection such as scrfd and retina, and then take a wider margin for both left and right, I don't know if this is the best way.
Thank you.

**Reply by leondgarse** [reply]
Uh, so you just wanna create your own dataset like glint360K one. I misunderstood you wanna do face detection in glint360K...
- Technically, faces extracted using `scrfd` / `retinaface` should be compatible with `Glint360K`. How your extracted yours?
- You can also refer to my [efficientnetV2_basic_test.ipynb](https://colab.research.google.com/drive/1vmAEfF9tUgK2gkrS5qVftadTyUcX343D?usp=sharing) the last `Keras insightface test` part. It's an example of using `scrfd` and also some instances of extracted images. I think the margin just alright.

**Reply by leondgarse** [reply]
I've uploaded the `YoloV5FaceDetector`, which is a pure TF one, and also the above colab ipynb script. The usage is almost the same:
```py
""" Previous """
# from eval_folder import Face_detection
# det = Face_detection()

""" Now """
import face_detector
det = face_detector.YoloV5FaceDetector()
# Or det = face_detector.SCRFD()
```
Other processes keep unchanged.

**Reply by nadongjin** [reply]
Thank you very much.
You're a great developer.
